unitTests/ui/FacetValueSuggestionsProviderTest.ts

```diff
@@ -111,5 +115,34 @@ export function FacetValueSuggestionsProviderTest() {
         done();
       });
     });
+
+    describe('with an additional expression', () => {
+      const anExpression = '@foofield==barvalue';
+
+      beforeEach(() => {
+        test = new FacetValueSuggestionsProvider(environment.queryController, {
+          field: <string>someField,
+          expression: anExpression
+        });
+      });
+
+      it(`should execute listFieldValuesBatch with value to search,
+          the additional expression and reference`, async done => {
+        await test.getSuggestions([valueToSearch]);
+
+        expect(environment.searchEndpoint.listFieldValuesBatch).toHaveBeenCalledWith({
+          batch: <IListFieldValuesRequest[]>[
+            {
+              ...baseFieldValueRequest,
+              queryOverride: `${anExpression} ${valueToSearch.text}`
+            },
+            {
+              field: someField
+            }
+          ]
+        });
+        done();
+      });
+    });
```

Comment:
On line 101 of `FacetValueSuggestionsProvider.ts`, I see a `.filter` method call to keep truthy values. Is there a test capturing the behaviour? If not, is the `.filter` needed?

Reply:
The previous existing tests cover the case when `expression` is falsy because it is not set. 

I took that way of conditionally concatenating the expression only if it's truthy from line 159. Before that I had a ternary like: 
 ```
this.buildListFieldValueRequest(
      [
          ...queryParts, 
          ...(this.options.expression ? [this.options.expression] : []),
          keyword.text
      ].join(' '))
```
And before that, a whole `if/else` statement calling `this.buildListFieldValueRequest` with different parameters. Do you think readability was sacrificed for the sake of less verbosity?

Reply:
I like the `.filter` approach, but perhaps using a temporary variable to store the result would be a little easier to read. e.g.

```
const expressions = this.options.expression.filter(exp => !!exp);
this.buildListFieldValueRequest([...queryParts, ...expressions, keyword.text].join(' '));
```